quaint/src/visitor/query_writer.rs

```diff
@@ -0,0 +1,33 @@
+use crate::Value;
+use query_template::{Fragment, QueryTemplate};
+use std::fmt;
+use std::fmt::Display;
+
+pub(crate) trait QueryWriter {
+    fn write_string_chunk<D: fmt::Display>(&mut self, s: D);
+    fn write_parameter(&mut self);
+    fn write_parameter_tuple(&mut self);
+}
+
+impl QueryWriter for QueryTemplate<Value<'_>> {
+    fn write_string_chunk<D: Display>(&mut self, value: D) {
+        match self.fragments.last_mut() {
+            Some(Fragment::StringChunk(chunk)) => {
+                chunk.push_str(&value.to_string());
+            }
+            _ => {
+                let mut chunk = String::with_capacity(4096);
```

Comment:
This feels a bit much for a chunk between parameters. Should we make it smaller, or call `shrink_to_fit` on the last chunk when pushing a new one (or both)?

Reply:
Calling `shrink_to_fit` sounds like a good idea.

I would not make the initial capacity smaller in order to avoid possible re-allocations and copying.

Reply:
Since `write_string_chunk` consumes its input `String` now, we can use that as initial buffer of any new `StringChunk` fragment without specifying a capacity. It may result in a few reallocations and copying, but for small buffers that should be fast enough (staying inside the CPU cache). 

Issuing `shrink_to_fit` on closing each `StringChunk` would complicate the interface. It may not be worth the premature optimization here.